### DISCUSSION TITLE
Can I customize the throbber using CSS?

### DISCUSSION THREAD

**Discussion by silverwings15** [mbnuqw/sidebery#1565] [Q&A] [answered]
Hi, I've been trying some old css snippets online to customize the throbber (_.tab-throbber::before_ and _.tab-throbber[busy]::before_) but none of them work as of 5.2.0. i'm trying to replace the default throbber with an external animation, and maybe replace the entire favicon of the loading page with the throbber instead of having it on top of the favicon. 

appreciate any help!

**Comment by silverwings15** [comment]
another question: can i use css to customize the indent of the active tab from the right, like so?

![image](https://github.com/mbnuqw/sidebery/assets/10742312/c34e1b85-b073-4e74-8bdd-c6aac6568241)

**Comment by mbnuqw** [comment]
> Can I customize the throbber using CSS?

Sure, it has [`.progress-spinner`](https://github.com/mbnuqw/sidebery/blob/1311c1cbe37975ccd8bf32aa7aa96f1ce6d47dce/src/sidebar/components/tab.vue#L48) class. Current styles: [tab.styl](https://github.com/mbnuqw/sidebery/blob/1311c1cbe37975ccd8bf32aa7aa96f1ce6d47dce/src/styles/themes/proton/sidebar/tab.styl#L410)

> can i use css to customize the indent of the active tab from the right, like so?

Try to play with the width of `.body`, something like `.Tab[data-active="true"] .body { width: calc(100% + var(--shift-for-active-tab))`; }.

**Reply by silverwings15** [reply]
sorry but i still need some hand holding here 😅

1. how do i disable the current styles completely so i can play with the new styles without e.g. being boxed into the bottom right quadrant of the favicon? also, how do i get rid of the transition to the black FF logo that indicates a loading tab?

2. i tried defining the variable like so

`#root.root {--shift-for-active-tab: 180px;}`

but your suggested code will shift the tab title to the right instead. if i try: `.Tab[data-active="true"] .body { width: 180px; }` then it works, but the effect isn't consistent because children tabs are already partially indented compared to parent tabs, so depending on the width of the sidebar, the deeper the children tab is nested, the less visible the effect becomes

![image](https://github.com/mbnuqw/sidebery/assets/10742312/610b80b4-6881-432d-91b3-a112e99eba29)


![image](https://github.com/mbnuqw/sidebery/assets/10742312/9ce7c6fb-d382-4611-8d58-226cf6146f58)



also, how come it's not possible to use `margin-right` for simplicity like in TST?

**Reply by silverwings15** [reply]
@mbnuqw managed to figure out the favicon/throbber thanks to the code you posted in https://github.com/mbnuqw/sidebery/issues/1586. my setup looks like this now (i think you have to click on the gif for it to play?):

![firefox_8tDSckEReF](https://github.com/mbnuqw/sidebery/assets/10742312/483e0343-73be-4f7f-a399-2b5fffe9ceac)

hope you can look into the other issue when you have the chance. thank you very much!

**Reply by mbnuqw** [reply]
> how do i disable the current styles completely

[`all` property](https://developer.mozilla.org/en-US/docs/Web/CSS/all)

> how do i get rid of the transition to the black FF logo that indicates a loading tab?

Not sure what transition did you mean, but you can go to [`tab.styl`](https://github.com/mbnuqw/sidebery/blob/v5/src/styles/themes/proton/sidebar/tab.styl) and check all transitions here, then use devtools to find appropriate selectors and play with values.

> Regarding shifted tab...

Post your full styles, and I'll check what's going on there and how can help

**Reply by silverwings15** [reply]
this is my current styles

```
#root.root {--general-border-radius: 8px;}
#root.root {--ctx-menu-fg: whitesmoke;}
#root.root {--tabs-activated-shadow: 0 1px 4px -1px rgba(0,0,0,0.282), inset 0 0 0 1px rgba(255,255,255,0.031);}
#root.root {--tabs-lvl-opacity: 0.16;}
#root.root {--ctx-menu-shadow: 0 0 0 1px rgba(255,255,255,0.086), 0 1px 16px 0 rgba(0,0,0,0.282);}
#root.root {--ctx-menu-separator: black;}
#root.root {--ctx-menu-border-radius: 5px;}
#root.root {--status-notice: #00e9fb;}
#root.root {--tabs-font: sans-serif;}
#root.root {--d-swift: 0.1s;}
#root.root {--tabs-activated-color-layer-opacity: 0.3;}
#root.root {--tabs-color-layer-opacity: 0.4;}
#root.root {--toolbar-height-personal: 38px;}

#root.root {--general-margin: 0.05px;}
#root.root {--tabs-discarded-title-opacity: 0.6;}

/* ----------- Tab title styles ----------- */

/* moves container indicator to the left */
.Tab .ctx { left: 1px; }

.Tab[data-discarded="true"] .title { 
    opacity: 75%;
}

.Tab[data-unread="true"] .title {font-style: italic;}


.Tab[data-pin="true"] .audio {
    top: 12px !important;
    left: 15px !important;
}

.Tab[data-parent="true"] .fav .exp {
 opacity: 0 !important; /* <- Hide parent tree arrow*/
}


/*--Bolds the active tab text--*/
.Tab[data-active="true"] .title {
  font-weight: bold;
  font-size: 13px;}

/*--New throbber--*/

#root .Tab[data-loading="true"] .fav > .fav-icon {
  opacity: 0;
}
#root .Tab[data-loading="true"] .fav {
  opacity: 1;
}
.Tab[data-loading="true"] .fav:before {
  content: url("https://i.imgur.com/qnTQWIM.png");
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  font-size: 1rem;
  line-height: 16px;
  text-align: center;
}

/*--Hide default throbber--*/
.Tab .progress-spinner {
display: none !important;
}
```
to indent the active tab from the right, i use the following code in TST: 

```
tab-item.active {
margin-right: 18px !important;
transition: scale 0.2s ease-in-out !important;
}
```
which indents any active tab a consistent 18px from the right, like so

![image](https://github.com/mbnuqw/sidebery/assets/10742312/f496ab42-7624-4449-a619-2526cfa625df)

if i try to replicate this in Sidebery using:

```.Tab[data-active="true"] .body { margin-right: 18px; }``` or 

```
#root.root {--shift-for-active-tab: 180px;}
.Tab[data-active="true"] .body { width: calc(100% + var(--shift-for-active-tab)); }
```
then it has no effect (last time i tried the second one, it shifted the tab title instead, weird)

if i try

```.Tab[data-active="true"] .body { width: 180px; }```

then it works, but the effect is not consistent like in TST 

![image](https://github.com/mbnuqw/sidebery/assets/10742312/767a5513-4378-4a72-a507-8d4d27fac075)

**Reply by mbnuqw** [reply, marked as answer]
Try this:
```css
.Tab[data-active="true"] {
  padding-right: 18px;
}
```

**Reply by silverwings15** [reply]
aw yiss, perfecto. thank you very much!!

**Reply by silverwings15** [reply]
@mbnuqw one more small request: can i make it so that this padding doesn't affect pinned tabs??
